include/lsst/afw/table/FieldBase.h

```diff
@@ -237,30 +241,50 @@ struct FieldBase<std::string> {
     static std::string getTypeString();
 
     /// @brief Return the number of subfield elements (equal to the size of the string,
-    ///        including a null terminator).
+    ///        including a null terminator), or 0 for a variable-length string.
     int getElementCount() const { return _size; }
 
     /// @brief Return the maximum length of the string, including a null terminator
-    ///        (equal to the number of subfield elements).
+    ///        (equal to the number of subfield elements), or 0 for a variable-length string.
     int getSize() const { return _size; }
 
+    /// Return true if the field is variable-length (each record can have a different size array).
+    bool isVariableLength() const { return _size == 0; }
 protected:
     /// Needed to allow Keys to be default-constructed.
     static FieldBase makeDefault() { return FieldBase(0); }
 
     /// Defines how Fields are printed.
     void stream(std::ostream &os) const { os << ", size=" << _size; }
 
-    /// Used to implement RecordBase::operator[] (non-const).
-    Reference getReference(Element *p, ndarray::Manager::Ptr const &m) const { return p; }
+    /// Used to implement BaseRecord::operator[] (non-const).
+    Reference getReference(Element *p, ndarray::Manager::Ptr const &m) const {
+        if (isVariableLength()) {
+            // can't be done until C++17; who would manage the memory?
```

Comment:
I'd have changed it the opposite way: it's clear this could be done with C++17 (at least as it's imagined today), but it's not about memory management.